### PR TITLE
Re-wired Source Control plug-in into IoC container

### PR DESCRIPTION
**Pull request by ckuhn203**
Step 1: Make it so we can open the window again. (Check!)
Step 2: Fix the buggy behavior we're been hearing about. (Next)
Step 3: ???
Step 4: Profit!

### CODE REVIEW

**Comment by INOPIAE** [comment]
Looks as if this patch will solve issue #766.

**Comment by Vogel612** [comment]
Additionally related: #832

File: RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs [hunk 1]
```diff
@@ -34,15 +30,13 @@ public override void Load()
...
             Bind<IMergeView>().To<MergeForm>();
 
-            //presenters
-            Bind<IChangesPresenter>().To<ChangesPresenter>();
-            Bind<IBranchesPresenter>().To<BranchesPresenter>();
-            Bind<ISettingsPresenter>().To<SettingsPresenter>();
-            Bind<IUnsyncedCommitsPresenter>().To<UnsyncedCommitsPresenter>();
+            //factories 
+            // todo: check on note below
+            // ninject is complaining about also having a SourceControlProviderFactoryProxy and a FolderBrowserFactoryProxy
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs]: The factory proxies are due to this method in the `RubberduckModule` class:

        // note convention: abstract factory interface names end with "Factory".
        private void ApplyAbstractFactoryConvention(IEnumerable<Assembly> assemblies)
        {
            _kernel.Bind(t => t.From(assemblies)
                .SelectAllInterfaces()
                .Where(type => type.Name.EndsWith("Factory"))
                .BindToFactory()
                .Configure(binding => binding.InSingletonScope()));
        }

Basically all abstract factories are automatically implemented by Ninject, which is great for the trivial implementations. Now, looking at that particular factory class, I *think* Ninject can handle it, but it would have to be tested; is there a binding for `ISourceControlProvider ~> GitProvider`?

**Comment by retailcoder** [comment]
@ckuhn203  I'm merging this one in, we'll just fix whatever broke if anything.

File: RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs (hunk 1 above)
Comment (retailcoder) [inline RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs]: Wait, no - the `assemblies` enumerable doesn't contain the `Rubberduck.SourceControl` assembly... not sure what's going on here.

**Comment by INOPIAE** [comment]
While testing the new implementation I run into this error when creating a new repository:

Ninject.ActivationException: Error activating ISourceControlProvider
No matching bindings are available, and the type is not self-bindable.
Activation path:
  1) Request for ISourceControlProvider

Suggestions:
  1) Ensure that you have defined a binding for ISourceControlProvider.
  2) If the binding was defined in a module, ensure that the module has been loaded into the kernel.
  3) Ensure you have not accidentally created more than one kernel.
  4) If you are using constructor arguments, ensure that the parameter name matches the constructors parameter name.
  5) If you are using automatic module loading, ensure the search path and filters are correct.

   bei Ninject.KernelBase.Resolve(IRequest request)
   bei Ninject.ResolutionExtensions.GetResolutionIterator(IResolutionRoot root, Type service, Func`2 constraint, IEnumerable`1 parameters, Boolean isOptional, Boolean isUnique)
   bei Ninject.Extensions.Factory.Factory.InstanceResolver.Get(Type type, String name, Func`2 constraint, IParameter[] parameters, Boolean fallback)
   bei Ninject.Extensions.Factory.StandardInstanceProvider.GetInstance(IInstanceResolver instanceResolver, MethodInfo methodInfo, Object[] arguments)
   bei Ninject.Extensions.Factory.FactoryInterceptor.Intercept(IInvocation invocation)
   bei Castle.DynamicProxy.AbstractInvocation.Proceed()
   bei Castle.Proxies.ISourceControlProviderFactoryProxy.CreateProvider(VBProject project)
   bei Rubberduck.UI.SourceControl.SourceControlPresenter.OnInitNewRepository(Object sender, EventArgs e) in d:\projects\rubberduck\Rubberduck\RetailCoder.VBE\UI\SourceControl\SourceControlPresenter.cs:Zeile 139.
   bei Rubberduck.UI.SourceControl.SourceControlPanel.RaiseGenericEvent(EventHandler`1 handler, EventArgs e) in d:\projects\rubberduck\Rubberduck\RetailCoder.VBE\UI\SourceControl\SourceControlPanel.cs:Zeile 110.
   bei Rubberduck.UI.SourceControl.SourceControlPanel.InitRepoButton_Click(Object sender, EventArgs e) in d:\projects\rubberduck\Rubberduck\RetailCoder.VBE\UI\SourceControl\SourceControlPanel.cs:Zeile 103.
   bei System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   bei System.Windows.Forms.ToolStripButton.OnClick(EventArgs e)
   bei System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   bei System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   bei System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   bei System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   bei System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   bei System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   bei System.Windows.Forms.Control.WndProc(Message& m)
   bei System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   bei System.Windows.Forms.ToolStrip.WndProc(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

File: RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs (hunk 1 above)
Comment (ckuhn203) [inline RetailCoder.VBE/UI/SourceControl/SourceControlBindings.cs]: No. No binding for `ISourceControlProvider` because it used to come from the factory. Looks like I've a bit more work to do here.

**Comment by ckuhn203** [comment]
@INOPIAE this was fixed in #852.
